Sources/URITemplate.swift

```diff
@@ -57,6 +57,12 @@ public struct URITemplate : CustomStringConvertible, Equatable, Hashable, Expres
     template = value
   }
 
+#if swift(>=4.0)
+  public init(from decoder: Decoder) throws {
```

Comment:
Would we also need to conform to `Decodable` to make this work? Perhaps it makes sense to break out the decodable behaviour as extension conformable like you did for `Codable`

Reply:
Codable acts as a composite type of `Encodable` and `Decodable` so we’re covered there. I’d like to break the initialiser out into an extension, but unfortunately Swift doesn’t let us declare initialisers in extensions.